nx/lib/nx.ex

```diff
@@ -3265,6 +3265,129 @@ defmodule Nx do
     end)
   end
 
+  @doc ~S"""
+  Split a tensor into train and test subsets.
+
+  `split` split must be an integer greater than zero and less than the length of the tensor.
+
+  ## Options
+
+    * `:axis` - The axis along which to take the values from. Defaults to `0`.
+
+  ## Examples
+
+    Split a tensor into two separate tensors.
+
+    iex> {train, test} = Nx.split(Nx.tensor([[3, 6, 5], [26, 75, 3], [23, 4, 1]]), 2, axis: 0)
+    iex> train
+    #Nx.Tensor<
+      s64[2][3]
+      [
+        [3, 6, 5],
+        [26, 75, 3]
+      ]
+    >
+    iex> test
+    #Nx.Tensor<
+      s64[1][3]
+      [
+        [23, 4, 1]
+      ]
+    >
+
+    iex> {train, test} = Nx.split(Nx.tensor([[3, 6, 5], [26, 75, 3], [23, 4, 1]]), 2, axis: 1)
+    iex> train
+    #Nx.Tensor<
+      s64[3][2]
+      [
+        [3, 6],
+        [26, 75],
+        [23, 4]
+      ]
+    >
+    iex> test
+    #Nx.Tensor<
+      s64[3][1]
+      [
+        [5],
+        [3],
+        [1]
+      ]
+    >
+
+    iex> {train, test} = Nx.split(Nx.tensor([[3, 6, 5, 20], [26, 75, 3, 9], [23, 4, 1, 5]], names: [:rows, :columns]), 2, axis: :columns)
+    iex> train
+    #Nx.Tensor<
+      s64[rows: 3][columns: 2]
+      [
+        [3, 6],
+        [26, 75],
+        [23, 4]
+      ]
+    >
+    iex> test
+    #Nx.Tensor<
+      s64[rows: 3][columns: 2]
+      [
+        [5, 20],
+        [3, 9],
+        [1, 5]
+      ]
+    >
+
+    iex>{train, test} = Nx.split(Nx.tensor([[3, 6, 5, 20], [26, 75, 3, 9], [23, 4, 1, 5]], names: [:rows, :columns]), 2, axis: :rows)
+    iex> train
+    #Nx.Tensor<
+      s64[rows: 2][columns: 4]
+      [
+        [3, 6, 5, 20],
+        [26, 75, 3, 9]
+      ]
+    >
+    iex> test
+    #Nx.Tensor<
+      s64[rows: 1][columns: 4]
+      [
+        [23, 4, 1, 5]
+      ]
+    >
+  """
+  @doc type: :indexed
+  def split(tensor, split, opts \\ [])
+
+  def split(%T{shape: shape} = tensor, split, opts) do
+    opts = keyword!(opts, axis: 0)
+    axis = Keyword.fetch!(opts, :axis)
+
+    if is_integer(split) and split > 0 do
+      axis_values = axes(tensor)
+      axis_names = names(tensor)
+
+      axis =
+        cond do
+          is_integer(axis) and axis in axis_values ->
+            axis
+
+          is_atom(axis) and axis in axis_names ->
+            dimensions = Enum.zip(axis_names, axis_values)
+            dimensions[axis]
+
+          true ->
+            raise ":axis is out of tensor bounds."
+        end
+
+      values = elem(shape, axis)
+      size = values - split
+
+      {
+        slice_along_axis(tensor, 0, split, axis: axis),
+        slice_along_axis(tensor, split, size, axis: axis)
+      }
+    else
+      raise "split must be an integer greater than zero and less than the length of the tensor."
+    end
```

Comment:
To support split as both integer and float, you can use a cond here instead of if